omniscidb/Tests/GroupByTest.cpp

```diff
@@ -327,13 +313,13 @@ class LowCardinalityThresholdTest : public ::testing::Test {
 };
 
 TEST_F(LowCardinalityThresholdTest, GroupBy) {
-  for (auto dt : {ExecutorDeviceType::CPU, ExecutorDeviceType::GPU}) {
-    SKIP_NO_GPU();
-
+  config().exec.heterogeneous.allow_cpu_retry = true;
+  for (auto dt : testedDevices()) {
     auto result = run_multiple_agg(
         R"(select fl,ar,dep from low_cardinality group by fl,ar,dep;)", dt);
     EXPECT_EQ(result->rowCount(), config().exec.group_by.big_group_threshold);
   }
+  config().exec.heterogeneous.allow_cpu_retry = false;
```

Comment:
Config is not going to be restored in case the test fails. We usually use `ScopeGuard` to restore config.